util/src/main/java/tc/oc/pgm/util/nms/NMSHacksProvider.java

```diff
@@ -0,0 +1,25 @@
+package tc.oc.pgm.util.nms;
+
+import tc.oc.pgm.util.bukkit.BukkitUtils;
+
+public interface NMSHacksProvider {
+
+  NMSHacks NMS_HACKS_CHOICE = chooseNMSHacks();
+
+  static NMSHacks chooseNMSHacks() {
+
+    try {
+      if (BukkitUtils.isSportPaper()) {
+        return new NMSHacksSportPaper();
+      } else {
+        return new NMSHacks1_8();
+      }
+    } catch (Throwable t1) {
+      return new NMSHacksNoOp();
+    }
+  }
```

Comment:
Adding some debug statement here posting which NMSHacks was chosen is probably also a good idea.

Reply:
Yeah I'll add something here to that effect